File: Easy/Java/575. Distribute Candies.java
class Solution {
    public int distributeCandies(int[] candyType) {
        Arrays.sort(candyType);
        int len = candyType.length;
        int count = 0;
        int limit = len / 2;

        for(int i = 0; i < len && 0 < limit; i++){
            while(i < len - 1 && candyType[i] == candyType[i + 1]){
                i++;
            }
            count++;
            limit--;
        }

        return count;
    }
}

File: Easy/JavaScript/575. Distribute Candies.js
var distributeCandies = function(candyType) {
    candyType.sort((a, b) => a - b);
    let len = candyType.length;
    let count = 0;
    let limit = len / 2;

    for(let i = 0; i < len && 0 < limit; i++){
        while(i < len - 1 && candyType[i] == candyType[i + 1]){
            i++;
        }
        count++;
        limit--;
    }

    return count;
};
